src/odemis/driver/xt_client.py

```diff
@@ -366,3 +365,362 @@ def set_beam_shift(self, x_shift, y_shift):
+        else:
+            self._focus = Focus(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._focus)
+
+    def terminate(self):
+        pass
+
+
+class Scanner(model.Emitter):
+    """
+    This is an extension of the model.Emitter class. It contains Vigilant
+    Attributes and setters for magnification, pixel size, translation, resolution,
+    scale, rotation and dwell time. Whenever one of these attributes is changed,
+    its setter also updates another value if needed e.g. when scale is changed,
+    resolution is updated, when resolution is changed, the translation is recentered
+    etc. Similarly it subscribes to the VAs of scale and magnification in order
+    to update the pixel size.
+    """
+
+    def __init__(self, name, role, parent, client, hfw_nomag, **kwargs):
+        model.Emitter.__init__(self, name, role, parent=parent, **kwargs)
+        self.client = client
+        self.parent = parent
+        self._hfw_nomag = hfw_nomag
+
+        self.dwellTime = model.FloatContinuous(
+            self.get_dwell_time(),
+            self.dwell_time_info()["range"],
+            unit=self.dwell_time_info()["unit"],
+            getter=self.get_dwell_time,
+            setter=self.set_dwell_time)
+
+        self.accelVoltage = model.FloatContinuous(
+            self.get_ht_voltage(),
+            self.ht_voltage_info()["range"],
+            unit=self.ht_voltage_info()["unit"],
+            getter=self.get_ht_voltage,
+            setter=self.set_ht_voltage
+        )
+
+        self.blanker = model.VAEnumerated(
+            self.beam_is_blanked(),
+            choices={True, False},
+            setter=self.blank_beam,
+            getter=self.beam_is_blanked)
+
+        self.spotSize = model.FloatContinuous(
+            self.get_ebeam_spotsize(),
+            self.spotsize_info()["range"],
+            unit=self.spotsize_info()["unit"],
+            getter=self.get_ebeam_spotsize,
+            setter=self.set_ebeam_spotsize)
+
+        self.beamShift = model.FloatContinuous(
+            self.get_beam_shift(),
+            self.beam_shift_info()["range"],
+            unit=self.beam_shift_info()["unit"],
+            getter=self.get_beam_shift,
+            setter=self.set_beam_shift)
+
+        fov = self.get_scanning_size()
+        self.horizontalFoV = model.FloatContinuous(
+            fov,
+            unit=self.scanning_size_info()["unit"],
+            range=self.scanning_size_info()["range"],
+            getter=self.get_scanning_size,
+            setter=self.set_scanning_size)
+
+        mag = self._hfw_nomag / fov
+        self.magnification = model.VigilantAttribute(mag, unit="",
+                                                     readonly=True)
+
+        selection_range = self.selected_area_info()["range"]
+        self._shape = selection_range[0][1], selection_range[1][1]
+        self.resolution = self.get_selected_area()[2:]
```

Comment:
`.resolution` should be a VA.
Be careful if you want to support it: there are typically a strong relationship between resolution/scale (and translation). For instance, if the user changes the scale, the resolution is automatically adjusted to keep the same FoV. See phenom or semcomedi.